code/hsec-tools/src/Security/Advisories/Format.hs

```diff
@@ -135,16 +136,25 @@ instance Toml.ToTable AdvisoryMetadata where
     ["aliases"   Toml..= amdAliases x | not (null (amdAliases x))] ++
     ["related"   Toml..= amdRelated x | not (null (amdRelated x))]
 
+instance Toml.FromValue GHCComponent where
+  fromValue v = case v of
+    Toml.Text' _ n
+      | Just c <- ghcComponentFromText n -> pure c
+    _ -> Toml.failAt (Toml.valueAnn v) "Invalid component, expected compiler|ghci|rts"
```

Comment:
This message is does not include all the options.  It might be worthwhile to define a lookup table that we can use for parsing and also for producing a list of the valid options.

Reply:
Oh well, we already have the conversion functions.